### ISSUE TITLE
No way to raise async errors in a reporter

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4057]
While axe-core allows for async reporters, if there is an async error axe doesn't surface that error properly when using the callback: 

```js
axe.addReporter('throwing', async (results, options, resolve, reject) => {
  await new Promise(r => setTimeout(r, 100))
  reject(new Error('Something went wrong'));
})

axe.run({ reporter: 'throwing' }, (err, results) => {
  results // This is a promise that will reject
});
```

By some fortunate accident in the code this does work when you're using `await axe.run({ reporter: 'throwing' })`.

**expected**: If a reporter throws an async error like happens above, that error gets passed to the `err` argument of the callback.

**Comment by padmavemulapati** [comment]
Valdiated using latest axe-core develop branch code  base,

`await axe.run({ reporter: 'throwing' })` it is generating report, not throwing any error





![Image](https://github.com/dequelabs/axe-core/assets/47409579/711c5f1b-d75d-4551-a6cc-76cb3b829832)

**Comment by straker** [comment]
@padmavemulapati you'll need to create a custom reporter that throws in order to validate that the reporter can cause axe-core to not return results but instead an error

```js
axe.addReporter('throwing', async (results, options, resolve, reject) => {
  setTimeout(() => reject('Reporter threw'), 100);
})

axe.run({ reporter: 'throwing' }, (err, results) => {
  console.log(err) // Reporter threw
  console.log(results)  // null
});
```
